app/controllers/webauthn_verifications_controller.rb

```diff
@@ -34,13 +38,20 @@ def authenticate
 
   private
 
+  def set_verification
+    @verification = WebauthnVerification.find_by(path_token: webauthn_token_param)
+
+    render_not_found and return unless @verification
+    redirect_to root_path, alert: t(".expired_or_already_used") if @verification.path_token_expires_at < Time.now.utc
```

Comment:
nit: `@verification.path_token_expires_at < Time.now.utc` could be a method in the WebauthnVerification model.